s-pipes-web/src/main/java/cz/cvut/spipes/rest/SPipesServiceController.java

```
@@ -221,7 +219,13 @@ private Model runModule(final Model inputDataModel, final MultiValueMap<String,
 
         ExecutionEngine engine = createExecutionEngine(configModel);
         ContextLoaderHelper.updateContextsIfNecessary(scriptManager);
-        Module module = PipelineFactory.loadModule(configModel.createResource(id));
+        Module module = null;
+        if (scriptManager.doesRegistryContainsEntity(id)) {
```

Why using once scriptManager and then PipelineFactory?

---

I'm trying to load it from the `scriptManager` if the required module exists there. If it does not, e.g. we have deleted it from the file. In this case, I'm trying to load it from the saved pipeline (so it is the last saved instance before deletion), so we would have at least some working module. I'm not sure if it is the best solution; maybe it would be better to let the code throw an exception in 
`if (module == null) {
    throw new SPipesServiceException("Cannot load module with id=" + id);
 }`
In case we are trying to load a non-existing module

---

Also, deleting an attempt to load with PiplelineFactory will cause test failures, so the tests should be changed in this case

---

I do not understand following expressions:
- "module exists there"  -- where? ... let's not talk about `scriptManager` without mentioning its role. Are you talking about cache or real state?
- "saved pipeline" -- confusing, i guess you are talking about cached instance of pipeline
- "would have at least some working module" -- not sure what working mean here. We want to have only current state of pipeline taken from the filesystem, nothing else.

> throw an exception in
> if (module == null) { throw new SPipesServiceException("Cannot load module with id=" + id); }
>In case we are trying to load a non-existing module

Yes, we want to throw the exception if the module is not within filesystem.

---

This is outdated.